kernel/apic.c

```diff
@@ -57,11 +158,99 @@ static uint8_t analyze_madt_entry (uint8_t *entry) {
 
 void apic_init (void) {
   vga_printf("CPU-Local APIC is at 0x%X\r\n", madt->local_interrupt_controller_address);
+  apic = (struct APIC*)phys_to_virt((uint64_t)madt->local_interrupt_controller_address);
 
   const uint32_t total_length = madt->h.Length;
   uint8_t *slice = (uint8_t *)madt;
   uint32_t offset = sizeof(struct MADT);
   while (offset < total_length) {
     offset += analyze_madt_entry(&slice[offset]);
   }
+
+  apic->spurious_interrupt_vector.val = 0xFF | APIC_SPURIOUS_VECTOR_ENABLE_APIC;
+}
+
+void ap_apic_init (void) {
```

Comment:
This isn't called afaict? and is duplicating the line above it?

Reply:
It's used in the upcoming CL, and seemed related enough to this diff to include it.